contracts/KSElasticLMV2.sol

```diff
@@ -132,18 +132,21 @@ contract KSElasticLMV2 is IKSElasticLMV2, KSAdmin, ReentrancyGuard {
     //deploy farmingToken if needed
     address destination;
     if (isUsingToken) {
-      bytes memory creationCode = abi.encodePacked(
-        farmingTokenCreationCode,
-        abi.encode(msg.sender)
-      );
+      bytes memory creationCode = farmingTokenCreationCode;
       bytes32 salt = keccak256(abi.encode(msg.sender, fId));
       assembly {
         destination := create2(0, add(creationCode, 32), mload(creationCode), salt)
         if iszero(extcodesize(destination)) {
           revert(0, 0)
         }
       }
+
       farm.farmingToken = destination;
+
+      //grant admin to msg.sender
+      IKyberSwapFarmingToken farmingToken = IKyberSwapFarmingToken(destination);
+
+      farmingToken.grantRole(farmingToken.DEFAULT_ADMIN_ROLE(), msg.sender);
```

Comment:
Should we use external call to get DEFAULT_ADMIN_ROLE() or just passed as 0x?

Reply:
Better to just fetch it, as we don't know if later we change the creationCode to another token with the same value of DEFAULT_ADMIN_ROLE